hazelcast/src/main/java/com/hazelcast/query/impl/predicates/InPredicate.java

```diff
@@ -63,11 +64,12 @@ protected boolean applyForSingleAttributeValue(Map.Entry entry, Comparable attri
         if (set == null) {
             set = createHashSet(values.length);
             for (Comparable value : values) {
-                set.add(convert(attributeValue, value));
+                Comparable converted = convert(attributeValue, value);
```

Comment:
That shouldn't be a problem since we always canonicalize the values in the set and a lookup value before checking for its presence in the set, so even if they are of different types initially canonical representation is still the same.